docs/cram-tests/index.md

```diff
@@ -0,0 +1,373 @@
+The proper fix is to instead add `(alias melange)` to `cram/deps`:
+
+```dune
+(cram
+ (deps
+  (alias melange)
+  ./output/src/cram-tests/SandwichTests.mjs))
+```
+
+Now the test finally fails. The presence of `(alias melange)` means that all
+build targets attached to the `@melange` alias are now dependencies for the cram
+tests in this directory.
+
+## `expand_aliases_in_sandbox` stanza
+
+If you kept the `$ npx tree` cram test around, you'll observe that even though
+all build targets attached to `@melange` are dependencies, they aren't copied
+into the sandbox directory. You can change that behavior by adding the
+[expand_aliases_in_sandbox
+stanza](https://dune.readthedocs.io/en/stable/dune-files.html#stanza-expand_aliases_in_sandbox)
+to your `dune-project` file:
+
+```dune
+; Copy all build targets for an alias into the sandbox
+
+(expand_aliases_in_sandbox)
+```
+
+Run `npm run test` again and you'll see that a lot of files have been copied
+over to the sandbox directory. Since `SandwichTests.mjs` is a `@melange` build
+target too, it actually no longer needs to be explicitly included in
+`cram/deps`:
+
+```
+(cram
+ (deps
+  (alias melange)))
+```
+
+Fix the bug in `Item.re` and re-run `npm run test`. Your cram test should pass
+once more.
+
+::: warning
+
+Using
+[expand_aliases_in_sandbox](https://dune.readthedocs.io/en/stable/dune-files.html#stanza-expand_aliases_in_sandbox)
+is more convenient, but it may noticeably impact cram test performance. If you
+feel your cram tests are too slow, you should remove it.
+
+:::
+
+---
+
+Fantastico! You've set up Dune cram tests to streamline the testing process. In
+the next chapter, we'll add logic for promotional discounts.
+
+## Overview
+
+- A cram test runs a command inside a sandbox and compares the output of that
+  command with some expected output
+- Dune has extensive support for cram tests, including the ability to promote
+  the latest output to become the expected output
+- A `cram` stanza is used to specify dependencies for your cram tests
+- Cram test files have the `.t` extension
+- Aliases are virtual build targets that other build targets can be attached to
+- Cram tests that produce nondeterministic output must have that be sanitized
+- Cram tests execute in a sandbox
+- For Melange projects, `(alias melange)` is generally the best dependency for
+  your cram tests
+- The `expand_aliases_in_sandbox` allows you to avoid having to put generated
+  `.mjs` files in your cram test dependencies, at the cost of having slower cram
+  tests
+
+## Exercises
+
+<b>1.</b> tbd
+
+<b>2.</b> tbd
+
+<b>3.</b> tbd
+
+-----
+
+View [source
+code](https://github.com/melange-re/melange-for-react-devs/blob/main/src/cram-tests/)
+and [demo](https://react-book.melange.re/demo/src/cram-tests/) for this chapter.
+
+-----
+
+[^1]: You might notice that the output from the Node test runner looks different
+    when run inside a cram test. That's because Note test runner uses different
```

Comment:
```suggestion
    when run inside a cram test. That's because Node test runner uses different
```

Reply:
Fixed in https://github.com/melange-re/melange-for-react-devs/pull/30/commits/d898b7d92a9e445b71a149f487a306cbdb348236